src/views/Checkout.vue

```diff
@@ -98,7 +100,38 @@ export default class Checkout extends Vue {
         await this.handleOnboardingResult();
 
         if (this.wallets.length === 0) this.goToOnboarding(true);
-        else this.getBalances();
+        else {
+            await this.getBalances();
+
+            // Handle optional sender address included in the request
+            if (this.request.sender) {
+                let errorMsg = '';
+                // Check if the address exists
+                const wallet = this.findWalletByAddress(this.request.sender.toUserFriendlyAddress(), true);
+                if (wallet) {
+                    // Check if that address has enough balance
+                    const accountOrContract = wallet.accounts.get(this.request.sender.toUserFriendlyAddress())
```

Comment:
would be cool to have a getter for this, so we could skip the wallet lookup.

Reply:
I don't understand. The wallet lookup in three lines above is essentially the check if the passed address exists at all and is even required, because we need the wallet id/keyId.

The only thing that could be simplified here is not looking up the AccountInfo/ContractInfo object again, but to look up the balance in the balance map used in `getBalances()`. I can do this by returning that map from the function.

Reply:
Ah, we need the wallet id, right. Fine then.